internal/core/model/adapters.rs

```diff
@@ -413,3 +415,510 @@ fn test_filter_model() {
     assert_eq!(filter.row_data(4).unwrap(), 8);
     assert_eq!(filter.row_count(), 5);
 }
+
+trait SortHelper<D> {
+    fn sort(&self, lhs: &D, rhs: &D) -> std::cmp::Ordering;
+}
+
+struct AscendingSortHelper;
+
+impl<D> SortHelper<D> for AscendingSortHelper
+where
+    D: std::cmp::Ord,
+{
+    fn sort(&self, lhs: &D, rhs: &D) -> std::cmp::Ordering {
+        lhs.cmp(rhs)
+    }
+}
+
+struct FnSortHelper<F, D>
+where
+    F: FnMut(&D, &D) -> std::cmp::Ordering + 'static,
+{
+    sort_function: RefCell<F>,
+    _type: PhantomData<D>,
+}
+
+impl<F, D> SortHelper<D> for FnSortHelper<F, D>
+where
+    F: FnMut(&D, &D) -> std::cmp::Ordering + 'static,
+{
+    fn sort(&self, lhs: &D, rhs: &D) -> std::cmp::Ordering {
+        (self.sort_function.borrow_mut())(lhs, rhs)
+    }
+}
+
+struct SortModelInner<M>
+where
+    M: Model + 'static,
+{
+    wrapped_model: M,
+    sort_helper: Box<dyn SortHelper<M::Data>>,
```

Comment:
It would be really nice to avoid the allocation and the dyn. In theory it should be possible, because we do know at compile time whether we're creating a sorted model where `T: Ord` or where the sort function is given.